www/docs/constructs/ViteStaticSite.md

```diff
@@ -0,0 +1,162 @@
+---
+description: "Docs for the sst.ViteStaticSite construct in the @serverless-stack/resources package"
+---
+
+The `ViteStaticSite` construct is a higher level CDK construct that makes it easy to create a Vite single page app. It provides a simple way to build and deploy the site to an S3 bucket; setup a CloudFront CDN for fast content delivery; and configure a custom domain for the website URL.
+
+It's designed to work with static sites built with [Vite](https://vitejs.dev/). It allows you to [automatically set environment variables](#configuring-environment-variables) in your Vite app directly from the outputs of your SST app. And for TypeScript apps, it can also create a `.d.ts` type definition file for the environments.
+
+The `ViteStaticSite` construct internally extends the [`StaticSite`](StaticSite.md) construct with the following pre-configured defaults.
+
+- [`indexPage`](StaticSite.md#indexpage) is set to `index.html`.
+- [`errorPage`](StaticSite.md#errorpage) is set to [`StaticSiteErrorOptions.REDIRECT_TO_INDEX_PAGE`](StaticSite.md#staticsiteerroroptions). So error pages are redirected to the index page.
+- [`buildCommand`](StaticSite.md#buildcommand) is `npm run build`.
+- [`buildOutput`](StaticSite.md#buildoutput) is the `dist` folder in your Vite app.
+- [`fileOptions`](StaticSite.md#fileoptions) sets the cache control to `max-age=0,no-cache,no-store,must-revalidate` for HTML files; and `max-age=31536000,public,immutable` for JS/CSS files.
+
+## Initializer
+
+```ts
+new ViteStaticSite(scope: Construct, id: string, props: ViteStaticSiteProps)
+```
+
+_Parameters_
+
+- scope [`Construct`](https://docs.aws.amazon.com/cdk/api/v2/docs/constructs.Construct.html)
+- id `string`
+- props [`ViteStaticSiteProps`](#vitestaticsiteprops)
+
+## Examples
+
+The `ViteStaticSite` construct is designed to make it easy to work with static sites built with [Vite](https://vitejs.dev/) or similar projects.
+
+### Creating a Vite app
+
+Deploys a Vite app in the `path/to/src` directory.
+
+```js
+new ViteStaticSite(this, "Site", {
+  path: "path/to/src",
+});
+```
+
+### Configuring environment variables
+
+The `ViteStaticSite` construct allows you to set the environment variables in your Vite app based on outputs from other constructs in your SST app. So you don't have to hard code the config from your backend. Let's look at how.
+
+Vite supports [setting build time environment variables](https://vitejs.dev/guide/env-and-mode.html). In your JS files this looks like:
+
+
+```js title="src/App.js"
+console.log(process.env.VITE_API_URL);
```

Comment:
@fwang nothing major, but I was just following through this and I think these are supposed to be `import.meta.env`